rs/ic_os/config/src/lib.rs

```diff
@@ -31,8 +74,170 @@ fn parse_config_line(line: &str) -> Option<(String, String)> {
 pub fn config_map_from_path(config_file_path: &Path) -> Result<ConfigMap> {
     let file_contents = read_to_string(config_file_path)
         .with_context(|| format!("Error reading file: {}", config_file_path.display()))?;
-    Ok(file_contents
+
+    let normalized_file_contents = file_contents.replace("\r\n", "\n").replace("\r", "\n");
+
+    Ok(normalized_file_contents
         .lines()
         .filter_map(parse_config_line)
         .collect())
 }
+
+fn is_valid_ipv6_prefix(ipv6_prefix: &str) -> bool {
+    ipv6_prefix.len() <= 19 && format!("{ipv6_prefix}::").parse::<Ipv6Addr>().is_ok()
+}
+
+pub fn get_config_ini_settings(config_file_path: &Path) -> Result<(NetworkSettings, bool)> {
+    let config_map: ConfigMap = config_map_from_path(config_file_path)?;
+
+    let ipv6_prefix = config_map
+        .get("ipv6_prefix")
+        .map(|prefix| {
+            // Prefix should have a max length of 19 ("1234:6789:1234:6789")
+            // It could have fewer characters though. Parsing as an ip address with trailing '::' should work.
+            if !is_valid_ipv6_prefix(prefix) {
+                bail!("Invalid IPv6 prefix: {}", prefix);
+            }
+            format!("{}::", prefix)
+                .parse::<Ipv6Addr>()
+                .context(format!("Failed to parse IPv6 prefix: {}", prefix))
+        })
+        .transpose()?;
+
+    // Per PFOPS - ipv6_subnet will never not be 64
+    let ipv6_subnet = 64_u8;
+    // Optional ipv6_address - for testing. Takes precedence over ipv6_prefix.
+    let ipv6_address = config_map
+        .get("ipv6_address")
+        .map(|address| {
+            // ipv6_address might be formatted with the trailing suffix. Remove it.
+
+            let address = address
+                .strip_suffix(&format!("/{}", ipv6_subnet))
+                .unwrap_or(address);
+            address
+                .parse::<Ipv6Addr>()
+                .context(format!("Invalid IPv6 address: {}", address))
+        })
+        .transpose()?;
+
+    if ipv6_address.is_none() && ipv6_prefix.is_none() {
+        bail!("Missing config parameter: need at least one of ipv6_prefix or ipv6_address");
+    }
+
+    let ipv6_gateway = config_map
+        .get("ipv6_gateway")
+        .context("Missing config parameter: ipv6_gateway")?
+        .parse::<Ipv6Addr>()
+        .context("Invalid IPv6 gateway address")?;
```

Comment:
I copied a lot of this code from the existing rs/ic_os/network/src/info.rs.

The old code will go away once we're fully switched over to using the new config struct.